cmd/mattermost/commands/config.go

```diff
@@ -320,6 +328,109 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 	}
 }
 
+func configResetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	currentConfig := app.Config()
+	defaultConfig := &model.Config{}
+	defaultConfig.SetDefaults()
+
+	if len(args) == 0 {
+		var confirmResetAll string
+		CommandPrettyPrintln("Are you sure you want to reset all the configuration settings?(YES/NO): ")
+		fmt.Scanln(&confirmResetAll)
+		if confirmResetAll == "YES" {
+			currentConfigMap := configToMap(*currentConfig)
+			for key := range currentConfigMap {
+				f := resetConfigValue(key, defaultConfig, currentConfig)
+				app.UpdateConfig(f)
+				newConfig := app.Config()
+				if err := newConfig.IsValid(); err != nil {
+					return err
+				}
+				if changed := config.FixInvalidLocales(newConfig); changed {
+					return errors.New("Invalid locale configuration")
+				}
+			}
+		}
+	}
+
+	for _, arg := range args {
+		f := resetConfigValue(arg, defaultConfig, currentConfig)
```

Comment:
Here we have the same problem, you can end up with a partial reset of your config, and actually, with an invalid configuration. Would be great if the changes to the config are done to a temporary config object, and at the end, we validate the new config and save it using the `app.SaveConfig` function.